.repo/sample-list-generator/src/api-detector.ts

```diff
@@ -0,0 +1,147 @@
+import { IApiItem, TApiTypeResult, TExtensionApiMap } from './types';
+import * as babel from '@babel/core';
+import fs from 'fs/promises';
+import { getAllFiles, singularize } from './utils';
+
+let EXTENSION_API_MAP: TExtensionApiMap = {};
+(async () => {
```

Comment:
Could we move this to a `loadExtensionApis` function maybe that gets called before anything else?

Reply:
I think we can remove the async function call as well and just have `loadExtensionApis()` in the top/bottom of the JS file. Then that function can update `EXTENSION_API_MAP`. Let me know if it would help for me to explain that more :)